crates/pico/src/memo.rs

```diff
@@ -98,22 +98,30 @@ fn update_derived_node(
     prev_value: &dyn DynEq,
     inner_fn: InnerFn,
 ) -> (Epoch, DidRecalculate) {
-    let (value, tracked_dependencies) =
-        with_dependency_tracking(db, derived_node_id.param_id, inner_fn);
-    let did_recalculate = if *prev_value != *value {
-        db.set_derive_node_time_updated(derived_node_id, tracked_dependencies.max_time_updated);
-        DidRecalculate::Recalculated
-    } else {
-        DidRecalculate::ReusedMemoizedValue
+    let (value, tracked_dependencies) = with_dependency_tracking(db, derived_node_id, inner_fn);
+    let did_recalculate = match value {
+        Some(v) => {
+            let did_recalculate = if *prev_value != *v {
+                db.set_derive_node_time_updated(
+                    derived_node_id,
+                    tracked_dependencies.max_time_updated,
+                );
+                DidRecalculate::Recalculated
+            } else {
+                DidRecalculate::ReusedMemoizedValue
+            };
+            db.insert_derived_node(
+                derived_node_id,
+                DerivedNode {
+                    dependencies: tracked_dependencies.dependencies,
+                    inner_fn,
+                    value: v,
+                },
+            );
+            did_recalculate
+        }
+        None => DidRecalculate::Error,
     };
-    db.insert_derived_node(
-        derived_node_id,
-        DerivedNode {
-            dependencies: tracked_dependencies.dependencies,
-            inner_fn,
-            value,
-        },
-    );
     (tracked_dependencies.max_time_updated, did_recalculate)
```

Comment:
If we receive `error`, `max_time_updated` is `Epoch::new()`. That seems like a sane default. But we should think about whether that's true.

Reply:
It makes sense even if it does not change overall execution. If we return error and register with wrong time_updated, we will call parent `inner_fn` which will call this memo again and will re-register with correct time_updated.